### ISSUE TITLE
Square Attack is not working

### ISSUE THREAD

**Issue by Code-Ranger** [Trusted-AI/adversarial-robustness-toolbox#925]
Hello, 
I am working on using square attacking (art.attacks.evasion.SquareAttack) based on the example code from https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/notebooks/classifier_blackbox.ipynb. While the modified code can execute, it seems the the attacking is not working (the result image has still been classified as original one).

I have tried different combination of parameters from https://github.com/max-andr/square-attack, it still does not working.
Could you provide some guidance on how to set the parameters (e.g., norm, eps, p_init, etc.) of SquareAttack for making it working?

Thanks very much!

**Comment by beat-buesser** [comment]
Hi @Code-Ranger Thank you very much for raising this question. Could you please provide more details about the model that you are evaluating? Would you be able to share a minimal example the we can run to reproduce your observations?

**Comment by Code-Ranger** [comment]
Hello, @beat-buesser ,
The major part of the code (modified from ART Hop Skip Jump example) is as following,
And the image I used for black-box attack is from ImageNet (please find it in the attachment)
![unicycle](https://user-images.githubusercontent.com/13929749/109903754-2970c000-7cd7-11eb-8413-b4963cf3ae52.jpg)


# load the pre-trained model
resnet = torchvision.models.resnet152(pretrained=True)
predict_counter = 0

# define sample predict function that reformats inputs, connects to wml scoring endpoint and
# returns one-hot encoded predictions
def predict(x):
    global predict_counter
    predict_counter += 1

    x = np.array(x[0])
    # convert np array back to image
    img = Image.fromarray(x.astype(np.uint8))
    img_t = preprocess(img)
    batch_t = torch.unsqueeze(img_t, 0)
    resnet.eval()
    out = resnet(batch_t)

    _, index = torch.max(out, 1)
    one_hot = np.zeros(out.shape[1])
    one_hot[index] = 1
    return one_hot

# Create blackbox object
classifier = BlackBoxClassifierNeuralNetwork(predict, image_array[0].shape, 1000, channels_first=False, clip_values=(0, 255))

# Select target image and check the prediction
target_image = image_array[0]
plt.imshow(Image.fromarray(target_image.astype(np.uint8)))
plt.show(block=False)
prediction = str(np.argmax(classifier.predict(image_array[:1], batch_size=image_array[0].shape[0]), axis=1)[0])
print("Prediction is: %s " % prediction, str(classes[prediction]))

# Generate HopSkipJump attack against black box classifier
attack = SquareAttack(estimator=classifier, max_iter=10000, eps=12.75, p_init=0.05)
iter_step = args.iter_step
x_adv = None
for i in range(args.round):
    x_adv = attack.generate(x=np.array([target_image]), x_adv_init=x_adv)

    prediction = str(np.argmax(classifier.predict(x_adv)[0]))
    print("Adversarial image at step %d." % (i * iter_step), "L2 error",
          np.linalg.norm(np.reshape(x_adv[0] - target_image, [-1])),
          "and class label is %s " % prediction, str(classes[prediction]))

    plt.imshow(Image.fromarray(x_adv[0].astype(np.uint8)))
    plt.show(block=False)

    attack.max_iter = iter_step
    print("Total number of predict() has been called: ", predict_counter)

**Comment by beat-buesser** [comment]
Hi @Code-Ranger It looks like your method `predict` is returning one-hot encoded predictions, but `SquareAttack` expects access to a model's logits predictions.

**Comment by Code-Ranger** [comment]
Hello, beat-busser,
Thanks. Now the square attack is working.
And I will close this issue. THX

**Comment by moazamjad** [comment]
@cyber-ranger Hi there. I was also trying the same as you are doing above but  the attack isn't running even though i am giving logits predictions. Can you please share your notebook as a sample that could help me?

**Comment by michaelasare1** [comment]
@cyber-ranger @beat-buesser @moazamjad Hi, I am trying to implement a square attack on an object detection model, specifically YOLO on any dataset. Can you please help me. Maybe sharing a notebook also. Thank you
